.github/workflows/e2e-ci.yml

```
@@ -0,0 +1,125 @@
+            echo "Waiting for frontend server... ($i)"
+            sleep 1
+          done
+
+      - name: Run Playwright E2E (smoke + preload test)
+        working-directory: terrafusion-cos/e2e
+        env:
+          CI: true
+          TF_FRONTEND_PORT: 3102
+          TF_API_5050_PORT: 5050
+        # Make Playwright strict so failures fail the job
+        run: |
```

actionlint (1.7.7)

shellcheck reported issue in this script: SC2046:warning:5:50: Quote this to prevent word splitting

(shellcheck)